tests/src/kakarot/test_kakarot.py

```diff
@@ -405,6 +408,182 @@ def test_failing_contract(self, cairo_run):
                 )
             assert not evm["reverted"]
 
+    class TestEthSendRawTransactionEntrypoint:
+        def test_should_raise_invalid_chain_id_tx_type_different_from_0(
+            self, cairo_run
+        ):
+            transaction = {
+                "type": 2,
+                "gas": 100_000,
+                "maxFeePerGas": 2_000_000_000,
+                "maxPriorityFeePerGas": 2_000_000_000,
+                "data": "0x616263646566",
+                "nonce": 34,
+                "to": "",
+                "value": 0x00,
+                "accessList": [],
+                "chainId": 9999,
+            }
+            tx_data = list(rlp_encode_signed_data(transaction))
+
+            with cairo_error(message="Invalid chain id"):
+                cairo_run(
+                    "test__eth_send_raw_transaction",
+                    tx_data_len=len(tx_data),
+                    tx_data=tx_data,
+                )
+
+        @SyscallHandler.patch("IAccount.get_nonce", lambda addr, data: [1])
+        @pytest.mark.parametrize("tx", TRANSACTIONS)
+        def test_should_raise_invalid_nonce(self, cairo_run, tx):
+            # explicitly set the nonce in transaction to be different from the patch
+            tx = {**tx, "nonce": 0}
+            tx_data = list(rlp_encode_signed_data(tx))
+            with cairo_error(message="Invalid nonce"):
+                cairo_run(
+                    "test__eth_send_raw_transaction",
+                    tx_data_len=len(tx_data),
+                    tx_data=tx_data,
+                )
+
+        @given(gas_limit=integers(min_value=2**64, max_value=DEFAULT_PRIME - 1))
+        def test_raise_gas_limit_too_high(self, cairo_run, gas_limit):
+            tx = {
+                "type": 2,
+                "gas": gas_limit,
+                "maxFeePerGas": 2_000_000_000,
+                "maxPriorityFeePerGas": 3_000_000_000,
+                "data": "0x616263646566",
+                "nonce": 34,
+                "to": "0x09616C3d61b3331fc4109a9E41a8BDB7d9776609",
+                "value": 0x5AF3107A4000,
+                "accessList": [],
+                "chainId": CHAIN_ID,
+            }
+            tx_data = list(rlp_encode_signed_data(tx))
+
+            with (
+                SyscallHandler.patch("IAccount.get_nonce", lambda _, __: [tx["nonce"]]),
+                cairo_error(message="Gas limit too high"),
+            ):
```

Comment:
note that if you put `nonce: 0` in the tx then you don't need to patch because 0 is the default storage read value

Reply:
`IAccount.get_nonce` it is an external call unfortunately, hence needs to be patched.